src/include/platform/internal/GenericDeviceInstanceInfoProvider.ipp

```diff
@@ -21,6 +21,18 @@ namespace chip {
 namespace DeviceLayer {
 namespace Internal {
 
+template <class ConfigClass>
+CHIP_ERROR GenericDeviceInstanceInfoProvider<ConfigClass>::GetDeviceName(MutableCharSpan & deviceNameSpan)
+{
+    constexpr char deviceName[] = CHIP_DEVICE_CONFIG_DEVICE_NAME;
+
+    ReturnErrorCodeIf(sizeof(deviceName) - 1 > deviceNameSpan.size(), CHIP_ERROR_BUFFER_TOO_SMALL);
+    memcpy(deviceNameSpan.data(), deviceName, sizeof(deviceName) - 1);
+    deviceNameSpan.reduce_size(sizeof(deviceName) - 1);
+
+    return CHIP_NO_ERROR;
```

Comment:
How about:
```suggestion
    return CopyCharSpanToMutableCharSpan(CharSpan::fromCharString(deviceName), deviceNameSpan);
```
This will deal with the extra null bit not being included, the size checks, etc.

Reply:
Great suggestion! I didn't know about this function thanks.